OP-780 - oh core make spring use xxx ACT (rebased)

goto-eof opened the pull request:
See OP-780.
Paired with https://github.com/informatici/openhospital-gui/pull/1334.

dbmalkovsky commented on sql/step_XX_drop_adm_deleted.sql:
```diff
@@ -0,0 +1 @@
+ALTER TABLE patient DROP COLUMN pat_deleted;
```
The file name looks wrong `step_XX....`    and the contents of this file is exactly the same as the next sql file

dbmalkovsky replied:
Note the file name: `step_XX_`  and the contents of this file is the same as the next: `ALTER TABLE patient ...`

dbmalkovsky commented on sql/step_XX_drop_pat_deleted.sql:
```diff
@@ -0,0 +1 @@
+ALTER TABLE patient DROP COLUMN pat_deleted;
```
Again the file name

dbmalkovsky commented on sql/step_04_all_following_steps.sql:
```diff
@@ -67,5 +67,7 @@ source step_70_increase_examination_note_field.sql;
 source step_72_hospital_changes.sql;
 source step_73_ward_visit_duration.sql;
+source step_72_drop_pat_deleted.sql;
+source step_73_drop_adm_deleted.sql;
```
By using step 72 and 73 you are replacing those that already exist there: `step_72_hospital_changes` and `step_73_ward_visit_duration`; please use other numbers.

mwithi commented on src/main/java/org/isf/patient/service/PatientIoOperationRepositoryImpl.java:
```diff
@@ -113,12 +112,10 @@ public List<Patient> getPatientsByParams(Map<String, Object> params) {
-		Predicate deletedN = cb.equal(patient.get("deleted"), "N");
-		Predicate deletedNull = cb.isNull(patient.get("deleted"));
-		Predicate notDeleted = cb.or(deletedN, deletedNull);
+		Predicate deletedN = cb.equal(patient.get("active"), Integer.valueOf(1));
```
I would rename into `notActive`

mwithi commented on src/test/java/org/isf/medicalstock/test/Tests.java:
```diff
@@ -245,12 +246,16 @@ public void testIoNewAutomaticDischargingMovementLotQuantityLessMovementQuantity
 		int code = setupTestMovement(false);
 		Movement foundMovement = movementIoOperationRepository.findById(code).get();
-
+System.out.println("ZZZ");
+System.out.println("1");
 		Medical medical = foundMovement.getMedical();
+		System.out.println("2");
 		MovementType movementType = foundMovement.getType();
 		Ward ward = foundMovement.getWard();
 		Supplier supplier = foundMovement.getSupplier();
+		System.out.println("3");
 		Lot lot2 = testLot.setup(medical, false); //we are going to create a second lot
+		System.out.println("4");
```
remove all println...

mwithi commented on sql/step_04_all_following_steps.sql:
```diff
@@ -69,3 +69,5 @@ source step_72_hospital_changes.sql;
 source step_73_ward_visit_duration.sql;
 source step_74_operation_report_button.sql;
 source step_75_add_prefix_to_table_names.sql;
+source step_76_drop_adm_deleted.sql;
+source step_77_drop_pat_deleted.sql;
```
I would combine into one single step, like `step_76_drop_deleted_fields_use_active`

mwithi reviewed (changes requested):
LGTM, anyway small changes are requested. Thanks!

mwithi commented:
it seems there are some conflicts too

dbmalkovsky commented on sql/step_04_all_following_steps.sql:
```diff
@@ -69,3 +69,5 @@ source step_72_hospital_changes.sql;
 source step_73_ward_visit_duration.sql;
 source step_74_operation_report_button.sql;
 source step_75_add_prefix_to_table_names.sql;
+source step_76_drop_adm_deleted.sql;
```
This needs to be renumbered again as there is a recently merged `step_76...`

mwithi commented:
> it seems there are some conflicts too

ok, just the script name and one import in the PatientExamination class

dbmalkovsky commented:
The `load_demo_data.sql` script needs to be updated too.

dbmalkovsky commented:
Why is the `SUP_DELETED` column in the `Supplier` object not included in this PR?   See this issue: https://openhospital.atlassian.net/browse/OP-527.  Doesn't it follow the same pattern as the other two?

mwithi commented:
> Why is the `SUP_DELETED` column in the `Supplier` object not included in this PR? See this issue: https://openhospital.atlassian.net/browse/OP-527. Doesn't it follow the same pattern as the other two?

yes, good catch! 
actually, the columns drop is a step that could have been done "just after" OP-780 and not "inside" it
the OP-527 is indeed "blocked by" OP-780...

goto-eof commented:
> Why is the `SUP_DELETED` column in the `Supplier` object not included in this PR? See this issue: https://openhospital.atlassian.net/browse/OP-527. Doesn't it follow the same pattern as the other two?



> > Why is the `SUP_DELETED` column in the `Supplier` object not included in this PR? See this issue: https://openhospital.atlassian.net/browse/OP-527. Doesn't it follow the same pattern as the other two?
> 
> yes, good catch! actually, the columns drop is a step that could have been done "just after" [OP-780](https://openhospital.atlassian.net/browse/OP-780) and not "inside" it the [OP-527](https://openhospital.atlassian.net/browse/OP-527) is indeed "blocked by" [OP-780](https://openhospital.atlassian.net/browse/OP-780)...

done.

dbmalkovsky commented on sql/step_81_drop_adm_deleted.sql:
```diff
@@ -0,0 +1 @@
+ALTER TABLE oh_admission DROP COLUMN pat_deleted;
```
Shouldn't the column that is being dropped be `adm_deleted`?  `pat_deleted` doesn't exist in the table.

dbmalkovsky commented:
Why isn't removing `SUP_DELETED` from `OH_SUPPLIER` not included in this PR?

goto-eof commented:
> Why isn't removing `SUP_DELETED` from `OH_SUPPLIER` not included in this PR?

it was a forgetfulness. I added the missing script, fixed the other one and tested them.

mwithi commented:
> > Why isn't removing `SUP_DELETED` from `OH_SUPPLIER` not included in this PR?
> 
> it was a forgetfulness. I added the missing script, fixed the other one and tested them.

As I was saying above, the deletion should be done in a single step.

mwithi commented:
Some conflicts to be resolved

dbmalkovsky commented on sql/step_04_all_following_steps.sql:
```diff
@@ -74,6 +74,4 @@ source step_77_laboratory_button_in_admission_patient.sql;
 source step_79_rename_pat_photo_field.sql;
 source step_80_add_ward_opd.sql;
-source step_81_drop_adm_deleted.sql;
-source step_82_drop_pat_deleted.sql;
```
Step 81 already is being used.

dbmalkovsky commented on src/main/java/org/isf/admission/service/AdmissionIoOperations.java:
```diff
@@ -264,7 +259,7 @@ public int getNextYProg(String wardId) throws OHServiceException {
 
 	public static LocalDateTime getNow() {
-		LocalDateTime now = TimeTools.getNow();
+		LocalDateTime now = LocalDateTime.now();
```
Please do not change this; the method `getNow()` is used so nanoseconds are not included in the time and thus can be used successfully across operating systems and architectures without issue.

goto-eof replied:
I made the rebase incorrectly

dbmalkovsky commented on src/main/java/org/isf/admission/service/AdmissionIoOperations.java:
```diff
@@ -232,15 +227,15 @@ public int getNextYProg(String wardId) throws OHServiceException {
 		} else {
-			first = now.with(firstDayOfYear()).with(LocalTime.MIN).truncatedTo(ChronoUnit.SECONDS);
-			last = now.with(lastDayOfYear()).with(LocalTime.MAX).truncatedTo(ChronoUnit.SECONDS);
+			first = now.with(firstDayOfYear()).with(LocalTime.MIN);
```
Please do not change this; see note at line 262

dbmalkovsky commented on src/main/java/org/isf/admission/service/AdmissionIoOperations.java:
```diff
@@ -232,11 +228,11 @@ public int getNextYProg(String wardId) throws OHServiceException {
 			if (now.getMonthValue() < Month.JUNE.getValue()) {
-				first = now.minusYears(1).withMonth(Month.JULY.getValue()).withDayOfMonth(1).with(LocalTime.MIN).truncatedTo(ChronoUnit.SECONDS);
-				last = now.withMonth(Month.JUNE.getValue()).withDayOfMonth(30).with(LocalTime.MAX).truncatedTo(ChronoUnit.SECONDS);
+				first = now.minusYears(1).withMonth(Month.JULY.getValue()).withDayOfMonth(1).with(LocalTime.MIN);
```
Please put back the `.truncatedTo()` methods in this line and through line 235

mwithi commented on src/main/java/org/isf/admission/service/AdmissionIoOperations.java:
```diff
@@ -124,9 +123,7 @@ public AdmittedPatient loadAdmittedPatient(final Integer patientId) {
-		if (isLoadPatientProfilePhotoFromDb) {
-			Hibernate.initialize(patient.getPatientProfilePhoto());
-		}
+		Hibernate.initialize(patient.getPatientProfilePhoto());
```
```suggestion
		if (isLoadPatientProfilePhotoFromDb) {
			Hibernate.initialize(patient.getPatientProfilePhoto());
		}
```

mwithi commented on src/main/java/org/isf/admission/service/AdmissionIoOperations.java:
```diff
@@ -40,7 +40,6 @@
 import org.isf.patient.model.Patient;
 import org.isf.patient.service.PatientIoOperationRepository;
-import org.isf.patient.service.PatientIoOperations;
 import org.isf.utils.db.TranslateOHServiceException;
```
```suggestion
import org.isf.patient.service.PatientIoOperations;
import org.isf.utils.db.TranslateOHServiceException;
```

mwithi commented on sql/step_82_drop_deleted.sql:
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE oh_admission DROP COLUMN adm_deleted;
+ALTER TABLE oh_patient DROP COLUMN pat_deleted;
+ALTER TABLE oh_supplier DROP COLUMN sup_deleted;
```
Sorry, just realized that:

The script, before to deleted those fields, should also "migrate" previous data, by setting `XXX_ACTIVE=0` for all `XXX_DELETED='Y'`.

`load_demo_data.sql` should also be provided.

mwithi commented:
Merging into a informatici repository
